samples/enforcePublicAccessPrevention.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This application demonstrates how to perform basic operations on buckets with
+ * the Google Cloud Storage API.
+ *
+ * For more information, see the README.md under /storage and the documentation
+ * at https://cloud.google.com/storage/docs.
+ */
+
+function main(bucketName = 'my-bucket') {
+  // [START storage_enforce_public_access_prevention]
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const bucketName = 'Name of a bucket, e.g. my-bucket';
+
+  // Imports the Google Cloud client library
+  const {Storage} = require('@google-cloud/storage');
+
+  // Creates a client
+  const storage = new Storage();
+
+  // Enforces public access prevention for the bucket
+  async function enforcePublicAccessPrevention() {
```

Comment:
Do we need to make this a separate function? If we can make `main` function `async`, then the code will just be like this:

```ts
async function main(.....) {
  ...
  // Enforces public access prevention for the bucket
  await storage.bucket(bucketName).setMetadata({.......});
  console.log(.....);
}
```
```

Reply:
@alexander-fenster It is not necessary to make it a separate function, but the way I have written it is how all the other samples in this repo are written. I can make this one different than the others if you think it is a better practice.

Reply:
It is very much necessary to make this a different function :) This is all covered here (internal):
https://g3doc.corp.google.com/company/teams/cloud-devrel/dpe/samples/nodejs.md?cl=head

Happy to chat about why this is what it is